text/0000-api-doc-frontpage-styleguide.md

```diff
@@ -0,0 +1,174 @@
+> default may be overridden. Logger implementations typically use the target to filter requests
+> based on some user configuration.
+
+There are several things that make this a good introduction section:
+    
+* A concise explanation of the crate's purpose.
+* The summary describes how the crate behaves in 3 short paragraphs.
+* Even if you don't want to read the rest of the `README`, you have enough to get started.
+* Readable by a layman, no jargon is introduced in the first few paragraphs.
+    
+What do we mean by "no jargon is introduced in the first few paragraphs"? Using the `rand` crate as
+an example - the crate's initial documentation probably should _not_ include a discussion of
+cryptographically secure random number generators since its primary purpose, for most Rust
+developers, will be to produce any random number. This information should instead be provided in a
+separate section that specifically discusses cryptographically secure random number generation.
+
+In general - it's good to give a developer a place to easily learn what your crate is about. Once
+they have the general idea, you can dive into more details that are specific to your crate,
+including jargon that would be common with its use.
+
+## First Example
+
+The first example should provide a concise example to help a developer understand how they can work
+with your crate. Ideally, this example should only demonstrate your crate and core Rust
+functionality. Avoid large examples with lots of features; larger examples are better fit for the
+`examples` directory.
+
+Let's look at a sample of a good first example of a crate. Notice that the author has focused on
+getting started, showing how to import and use the crate, and a few simple uses of common
+functionality.
+
+```rust
+#[macro_use] 
+extern crate log;
+
+pub fn shave_the_yak(yak: &Yak) {
+    info!(target: "yak_events", "Commencing yak shaving for {:?}", yak);
+
+    loop { 
+        match find_a_razor() { 
+            Ok(razor) => { 
+                info!("Razor located: {}", razor);
+                yak.shave(razor); 
+                break; 
+            } 
+            Err(err) => { 
+                warn!("Unable to locate a razor: {}, retrying", err); 
+            }
+        }
+    }
+}
+```
+
+## Crate Capabilities
+
+The core capabilities of a crate are the main reason people will use your crate.  In the next
+section, you can document what each of these core capabilities are and how to use them.  For
+example, in a crate about random numbers, you may have sections about: generating random numbers,
+fitting numbers to statistical distributions, use in cryptography, thread-local random numbers, and
+so on.
+
+It's helpful to introduce and give a clear description for each capability separately to help your
+readers understand each concept individually before they begin to combine capabilities.
+
+Here's a good example from the 'rand' crate:
```

Comment:
I wonder if this should be a link. When mentioning the log crate above you make it a link. I don't think a link has been provided for rand yet despite multiple mentions.